plugin/storage/es/samplingstore/storage.go

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) 2024 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package samplingstore
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/olivere/elastic"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/cmd/collector/app/sampling/model"
+	"github.com/jaegertracing/jaeger/pkg/es"
+	"github.com/jaegertracing/jaeger/plugin/storage/es/samplingstore/dbmodel"
+)
+
+const (
+	samplingIndex        = "jaeger-sampling-"
+	throughputType       = "throughput-sampling"
+	probabilitiesType    = "probabilities-sampling"
+	indexPrefixSeparator = "-"
+)
+
+type SamplingStore struct {
+	client                         func() es.Client
+	logger                         *zap.Logger
+	samplingIndexPrefix            string
+	indexDateLayout                string
+	maxDocCount                    int
+	samplingIndexRolloverFrequency time.Duration
+	adaptiveSamplingLookback       time.Duration
+}
+
+type SamplingStoreParams struct {
+	Client                         func() es.Client
+	Logger                         *zap.Logger
+	IndexPrefix                    string
+	IndexDateLayout                string
+	MaxDocCount                    int
+	SamplingIndexRolloverFrequency time.Duration
+	AdaptiveSamplingLookback       time.Duration
```

Comment:
there's no need to keep referring to 'sampling' here, it's already clear contextually (unlike es/config/Config where all kinds of attributes are mixed in one place)

```suggestion
	IndexRolloverFrequency time.Duration
	Lookback       time.Duration
	MaxDocCount                    int
```